contrib/guix/guix-build

```diff
@@ -2,22 +2,26 @@
 export LC_ALL=C
 set -e -o pipefail
 
+# Source the common prelude, which:
+#   1. Checks if we're at the top directory of the Bitcoin Core repository
+#   2. Defines a few common functions and variables
+#
+# shellcheck source=libexec/prelude.bash
+source "$(dirname "${BASH_SOURCE[0]}")/libexec/prelude.bash"
+
+
 ###################
-## Sanity Checks ##
+## SANITY CHECKS ##
 ###################
 
 ################
-# Check 1: Make sure that we can invoke required tools
+# Required non-builtin commands should be invokable
 ################
-for cmd in git make guix cat mkdir curl; do
-    if ! command -v "$cmd" > /dev/null 2>&1; then
-        echo "ERR: This script requires that '$cmd' is installed and available in your \$PATH"
-        exit 1
-    fi
-done
+
+check_tools cat mkdir make git guix
```

Comment:
`cat` and `git` already checked in the sourced `libexec/prelude.bash`.
Maybe combine both `check_tools` into one?

Reply:
The check is quite cheap, and only attest to the tools being used in the current script. Instead of trying to figure out how to optimize it, I think we can just leave it be if it's redundant. 😄